packages/status-react/src/protocol/use-sorted-chats.tsx

```diff
@@ -0,0 +1,73 @@
+import { useMemo } from 'react'
+
+import { useProtocol } from './provider'
+
+import type { Community } from '@status-im/js'
+
+export type Chat = Community['description']['chats'][0] & {
+  id: string
+}
+
+export type Category = {
+  id: string
+  name: string
+  position: number
+  chats: Chat[]
+}
+
+type Result = {
+  categories: Category[]
+  chats: Chat[]
+}
+
+function sortByPosition<T extends { position: number }>(items: T[]): T[] {
+  items.sort((a, b) => {
+    if (a.position < b.position) {
+      return -1
+    }
+    if (a.position > b.position) {
+      return 1
+    }
+    return 0
+  })
+
+  return items
+}
+
+export const useSortedChats = (): Result => {
```

Comment:
> Desktop currently adds categories to the bottom of the list

to me it makes sense, but I'm not picking favorites

Reply:
My reasoning being, for example, having categories per teams and no category for "prominent" chats like #general. So if I'm not a member of any team I don't have to scroll to the bottom past all the team specific chats to get to #general.

Also, having non-categorized chats first doesn't require an extra category like "general" containing a "#general" chat and being reordered to the top.